app/shared/src/GeolocationContext.js

```diff
@@ -0,0 +1,66 @@
+// @flow
+
+import * as React from 'react';
+
+const defaultState = {
+  lat: null,
+  lng: null,
+  canGetUserLocation: true,
+};
+
+const { Consumer, Provider: ContextProvider } = React.createContext({
+  ...defaultState,
+  actions: {
+    updateGeolocation: () => {},
+  },
+});
+
+type Props = {|
+  children: React.Node,
+|};
+
+type State = {|
+  lat: number | null,
+  lng: number | null,
+  canGetUserLocation: boolean,
+  actions: {|
+    updateGeolocation: () => void,
+  |},
+|};
+
+class Provider extends React.Component<Props, State> {
+  state = {
+    ...defaultState,
+    actions: {
+      updateGeolocation: this.updateGeolocation,
```

Comment:
This function is not used outside the context so why is it exposed? It's also `undefined` at this moment...

Reply:
the idea was to offer a way for components to refresh lat, lng if needed. Why it is undefined, I don't know 🙈

Reply:
> the idea was to offer a way for components to refresh lat, lng if needed